src/AlarmClock-Spec/AlarmClockPresenter.class.st

```diff
@@ -0,0 +1,58 @@
+"
+A presenter for an instance of AlarmClock.
+"
+Class {
+	#name : #AlarmClockPresenter,
+	#superclass : #SpPresenter,
+	#instVars : [
+		'model',
+		'elapsedTimeLabel',
+		'finishedLabel'
+	],
+	#category : #'AlarmClock-Spec'
+}
+
+{ #category : #specs }
+AlarmClockPresenter class >> defaultSpec [
+	^ SpBoxLayout newVertical
+		add: #elapsedTimeLabel;
+		add: #finishedLabel;
+		yourself
+]
+
+{ #category : #examples }
+AlarmClockPresenter class >> example1 [
+	self new
+		model: (AlarmClock new pass: 5 minutes; yourself);
+		openWithSpec
+]
+
+{ #category : #accessing }
+AlarmClockPresenter >> elapsedTimeString [
+	^ model elapsedTime asString
+]
+
+{ #category : #accessing }
+AlarmClockPresenter >> finishedString [
+	^ model finished
+		ifTrue: [ 'FINISHED' ]
+		ifFalse: [ 'NOT FINISHED' ]
```

Comment:
Uppercase?

Reply:
It's fine for the moment

Reply:
In the next iteration we can completely rewrite this presenter, anyway.

Reply:
I agree

Reply:
I totally agree

Reply:
I really totally agree

Reply:
I totally agree

Reply:
I totally agree

Reply:
I totally agree

Reply:
dasds

Reply:
ggggggggg

Reply:
ggggggggg

Reply:
ddd

Reply:
eee

Reply:
end

Reply:
end2

Reply:
end3

Reply:
end4